notifier/notifier.go

```diff
@@ -221,43 +220,118 @@ func (n *Notifier) collectRogueCallbacks() {
 	}
 }
 
-// Notify posts callback info to j.CallbackURL
-func (n *Notifier) Notify(j *job.Job) error {
+// MonitorDeliveries starts as many go routines as limit instructs in order to
+// poll/monitor the delivery reports channel.
+func (n *Notifier) MonitorDeliveries(ctx context.Context, limit int, dwg *sync.WaitGroup) {
+	dwg.Add(limit)
+
+	for i := 0; i < limit; i++ {
+		go n.monitorDeliveriesRoutine(ctx, dwg)
+	}
+}
```

Comment:
IMHO this function should just be inlined, since it's used only in 1 place. Right now the level of indirection it adds makes the code harder to follow.